### ISSUE TITLE
Local CI build prompts twice when requesting build information

### ISSUE THREAD

**Issue by davidwendt** [rapidsai/gha-tools#100]
Following [Reproducing CI Locally](https://docs.rapids.ai/resources/reproducing-ci/) instructions recently has started to prompt twice for the `RAPIDS_BUILD_TYPE`, `RAPIDS_REPOSITORY`, and `RAPIDS_REF_NAME`.

Example running `ci\build_docs.sh` (though any script that requires artifacts also does this):
```
...
[rapids-download-conda-from-s3] Local run detected.
[rapids-download-conda-from-s3] NVIDIA VPN connectivity is required to download workflow artifacts.

Enter workflow type (one of: pull-request|branch|nightly): pull-request

Suppress this prompt in the future by setting the 'RAPIDS_BUILD_TYPE' environment variable:
export RAPIDS_BUILD_TYPE=pull-request


Enter org/repository name (e.g. rapidsai/cudf): rapidsai/cudf

Suppress this prompt in the future by setting the 'RAPIDS_REPOSITORY' environment variable:
export RAPIDS_REPOSITORY=rapidsai/cudf


Enter pull-request number (e.g. 1546): 15375

Suppress this prompt in the future by setting the 'RAPIDS_REF_NAME' environment variable:
export RAPIDS_REF_NAME=pull-request/15375


Using HEAD commit for artifact commit hash. Overwrite this by setting the 'RAPIDS_SHA' environment variable:
export RAPIDS_SHA=77...c9

[rapids-download-from-s3] Downloading and decompressing https://downloads.rapids.ai/ci/cudf/pull-request/15375/77337ac/cudf_conda_cpp_cuda12_x86_64.tar.gz into /tmp/cpp_channel
[rapids-download-conda-from-s3] Local run detected.
[rapids-download-conda-from-s3] NVIDIA VPN connectivity is required to download workflow artifacts.

Enter workflow type (one of: pull-request|branch|nightly):

```
One prompt is from `rapids-download-conda-from-s3` and the 2nd is from `rapids-download-from-s3`.
Both export the variables so I'm not sure why the 2nd one does not see them.

After filling out both sets of prompts, the process completes successfully.
Simply workaround is to set these 3 variables locally before launching the script.

**Comment by vyasr** [comment]
Yeah I run into this sometimes too. The problem is that the tools aren't actually exporting those variables, they just forward them along via calls. I'd support setting them the first time they're provided to prevent this kind of thing unless there's a problem with that approach that I'm not aware of.

**Comment by jameslamb** [comment]
## Short Summary

I think this is resolved by recent work to move artifacts from S3 to the GitHub Actions artifact store, including https://github.com/rapidsai/docs/pull/601

Given something like this:

```shell
CPP_CHANNEL=$(rapids-download-conda-from-github cpp)
PYTHON_CHANNEL=$(rapids-download-conda-from-github python)
```

You'll be prompted for the inputs pull `RAPIDS_REPOSITORY` only once per each call to `rapids-download-conda-from-github`... in @davidwendt 's initial report, it was **TWICE** per call to `rapids-download-conda-from-s3`.

### Details

Reproducing `ci/build_docs.sh` from a recent `cudf` PR, following https://docs.rapids.ai/resources/reproducing-ci/#reproducing-ci-jobs

* PR: https://github.com/rapidsai/cudf/pull/19058
* build link: https://github.com/rapidsai/cudf/actions/runs/15392742690/job/43307811087#step:11:10

```shell
git clone \
  --branch branch-25.08 \
  git@github.com:rapidsai/cudf.git

cd ./cudf

docker run \
  --rm \
  --gpus 1 \
  --env RAPIDS_SHA=9ff4a06bde91328841d77f13e8ae0368d15eb297 \
  --pull=always \
  --volume $PWD:/repo \
  --workdir /repo \
  -it rapidsai/ci-conda:latest

ci/build_docs.sh
```

NOTE: I passed `RAPIDS_SHA` to prevent it from using the head commit on my checkout of `cudf`, as is done by default here:

https://github.com/rapidsai/gha-tools/blob/ec1f9ccce4fbf7e01077ee2c26a4bc1a661b62d4/tools/rapids-prompt-local-repo-config#L66-L71

That SHA I passed is the final commit on https://github.com/rapidsai/cudf/pull/19058

This prompted 1 time for each input and for me to authenticate with GitHub, which I did via a browser window.

Inputs I provided:

* workflow type: `pull-request`
* org/repository name: `rapidsai/cudf`
* pull-request number: `19058`

<img width="1007" alt="Image" src="https://github.com/user-attachments/assets/149bfdff-7335-43d6-a12b-61eb1824b3ea" />

I was then prompted a second time when it moved on to downloading the Python packages... but notably NOT prompted for GitHub auth again.

<img width="1009" alt="Image" src="https://github.com/user-attachments/assets/d00e6602-52e6-4713-926f-1e6e8e8a7c41" />

### Can we get this down to even fewer prompts?

I don't think we should.

Responding to @vyasr 's suggestion:

> *the tools aren't actually exporting those variables, they just forward them along via calls. I'd support setting them the first time they're provided to prevent this kind of thing*

This would require some changes across RAPIDS. Namely, instead of invoking these scripts like this in `ci/build_*.sh`:

```shell
CPP_CHANNEL=$(rapids-download-conda-from-github cpp)
PYTHON_CHANNEL=$(rapids-download-conda-from-github python)
```

They'd have to be `source`'d. Otherwise, there's no way they could "export those variables"... a child process cannot influence the environment variables of a parent process.

`source`-ing these scripts would introduce new risks and complexity. For example, we'd have to be more careful about collision in naming (since variable assignments would now happen in the same scope as `build_*.sh` and other calls to `rapids-*` scripts).

I think the current state is a pretty good one to be in. Namely:

* if you don't set the `RAPIDS_*` environment variables, you will get prompted once per call to `rapids-download-{conda,wheels}-from-github`
* you can avoid all those prompts via setting environment variables, as is noted by log messages:
  - https://github.com/rapidsai/gha-tools/blob/ec1f9ccce4fbf7e01077ee2c26a4bc1a661b62d4/tools/rapids-prompt-local-repo-config#L43-L44
* the variables are all available in CI logs, to make copying and pasting easier
  - ref: https://github.com/rapidsai/cudf/actions/runs/15392742690/job/43307811087#step:11:10

<img width="730" alt="Image" src="https://github.com/user-attachments/assets/f21f8dc0-cdf6-43b8-92aa-f3ecefddef6c" />

To clarify that point... this does not involve any interactive prompts about `RAPIDS_*` variables, and only 1 total GitHub auth prompt (which is persisted across the rest of the session):

```shell
docker run \
  --rm \
  --gpus 1 \
  --env RAPIDS_BUILD_TYPE=pull-request \
  --env RAPIDS_REF_NAME=pull-request/19058 \
  --env RAPIDS_REPOSITORY=rapidsai/cudf \
  --env RAPIDS_SHA=9ff4a06bde91328841d77f13e8ae0368d15eb297 \
  --pull=always \
  --volume $PWD:/repo \
  --workdir /repo \
  -it rapidsai/ci-conda:latest

ci/build_docs.sh
```
